homeassistant/components/shelly/__init__.py

```diff
@@ -710,7 +711,10 @@ async def _async_update_data(self) -> None:
     @property
     def model(self) -> str:
         """Model of the device."""
-        return cast(str, self.entry.data["model"])
+        try:
+            return cast(str, self.entry.data["model"])
+        except KeyError as err:
+            raise ConnectionError(FW_PROVISIONING_ISSUE_MSG) from err
```

Comment:
I don't think we should add this check.
Here we use the the config entry data and we can't get here if the device wasn't added properly (e.g. if model key was missing)

Reply:
What if the device is changed after setup ? For example a replacement for a hardware fault ?

Simone

Reply:
Than it will be identified as a new device (new MAC, new shellyid and new config entry)